py/selenium/webdriver/remote/remote_connection.py

```diff
@@ -103,41 +103,58 @@ def get_remote_connection_headers(cls, parsed_url, keep_alive=False):
 
         return headers
 
-    def __init__(self, remote_server_addr, keep_alive=False, resolve_ip=True):
+    # todo See if this actually does anything - Kamaroyl
+    @staticmethod
+    def parse_remote_server(remote_server_address, resolve_ip):
+        """If resolve ip is true, take apart remote_server_address and try to connect to address
+        unless the scheme is https, in which case just put the address back together
+        :param remote_server_address: address of remote server
+        :param resolve_ip: boolean on whether to try to resolve the hostname
+        :return: re-parsed remote_server_address
+        """
+        if resolve_ip:
+            parsed_url = parse.urlparse(remote_server_address)
+            if parsed_url.hostname:
+                port = parsed_url.port or None
+
+                if parsed_url.scheme == "https":
+                    ip = parsed_url.hostname
+
+                elif port and not common_utils.is_connectable(port, parsed_url.hostname):
+                    ip = None
+                    LOGGER.warning('Could not connect to port {} on host '
+                                   '{}'.format(port, parsed_url.hostname))
+
+                else:
+                    ip = common_utils.find_connectable_ip(parsed_url.hostname,
+                                                          port=port)
+                if ip:
+                    netloc = ip
+                    if parsed_url.port:
+                        netloc = common_utils.join_host_port(netloc,
+                                                             parsed_url.port)
+                    if parsed_url.username:
+                        auth = parsed_url.username
+                        if parsed_url.password:
+                            auth += ':%s' % parsed_url.password
+
+                        netloc = '%s@%s' % (auth, netloc)
+
+                    url_parts = (parsed_url.scheme, netloc, parsed_url.path,
+                                 parsed_url.params, parsed_url.query, parsed_url.fragment)
+                    remote_server_address = parse.urlunparse(url_parts)
+                else:
+                    LOGGER.warning('Could not get IP address for host: {}'
+                                   .format(parsed_url.hostname))
+        return remote_server_address
+
+    def __init__(self, remote_server_addr, keep_alive=False, retry_count=3, resolve_ip=True):
```

Comment:
I would set the retry_count to 0. Even though urllib3 only retries on connections and not on completed connections, we can't rely on this fact because a driver or remote end may repeat a command.

Reply:
I would make retry a class variable to set. Most users should rely on 0 retries.